### PR TITLE
ENH move to vs2017

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -306,8 +306,6 @@ pin_run_as_build:
     max_pin: x.x
   tiledb:
     max_pin: x.x.x
-  vc:
-    max_pin: x
```
Comment (beckermr) [inline recipe/conda_build_config.yaml]: @isuruf I removed this max pin since it appears to be wrong and covered by the run exports.

Comment (isuruf) [reply recipe/conda_build_config.yaml]: Yes, it's covered by run exports. Btw, `x` is correct, right?

Comment (beckermr) [reply recipe/conda_build_config.yaml]: I always get confused by this syntax. We set max_pin=None in the run exports, not 'x' IIRC. Doesn't that mean no entry here?

Comment (beckermr) [reply recipe/conda_build_config.yaml]: https://github.com/conda-forge/vc-feedstock/blob/master/recipe/meta.yaml#L34

Comment (isuruf) [reply recipe/conda_build_config.yaml]: We set `max_pin=x` at https://github.com/conda-forge/vc-feedstock/blob/master/recipe/meta.yaml#L33

Comment (beckermr) [reply recipe/conda_build_config.yaml]: Ack wrong line. Yes 'x' is right.

**Comment by dhirschfeld** [comment]
Just curious why you wouldn't move to vs2019?

**Comment by isuruf** [comment]
Because then we force downstream users to use VS 2019.

**Comment by dhirschfeld** [comment]
I thought the ucrt was suppose to enable you to build with other versions of Visual Studio?

i.e. even though Python was built with vs2015 you could still build extensions with a newer Visual Studio

**Comment by isuruf** [comment]
Sure, but not the other way around. You can't build upstream package with VS2019 and link it downstream with VS2017.

**Comment by dhirschfeld** [comment]
Thanks, that subtlety eluded me but is good to be aware of!

File: recipe/conda_build_config.yaml [hunk 2]
```diff
@@ -593,8 +591,8 @@ tk:
   - 8.6                # [not ppc64le]
 tiledb:
   - 1.7.0
-vc:                    # [win]
-  - 14                 # [win]
+vc:                      # [win]
+  - 14.1                 # [win]
```
Comment (beckermr) [inline recipe/conda_build_config.yaml:595]: @isuruf should we keep this at 14?

**Comment by beckermr** [comment]
@isuruf @conda-forge/core This one is ready for review and merging. I have one question above.

**Comment by beckermr** [comment]
@conda-forge-admin rerender

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (beckermr) [reply recipe/conda_build_config.yaml:595]: 14.1 doesn’t match the pinning and conda will pickup the later version anyways.

**Comment by CJ-Wright** [comment]
This seems reasonable to me but I'm not certain I have the needed knowledge.

**Comment by beckermr** [comment]
bump here @conda-forge/core 

I'd like to pull the trigger on this today and we have gotten one question about why it hasn't changed per the deadline yesterday.

File: recipe/conda_build_config.yaml (hunk 2 above)
Comment (isuruf) [reply recipe/conda_build_config.yaml:595]: Yes, let's keep it at 14

Comment (beckermr) [reply recipe/conda_build_config.yaml]: done

**Comment by isuruf** [comment]
We'll need to make a concurrent conda-smithy release that bumps the windows image.

**Comment by isuruf** [comment]
I mean the windows image for appveyor

**Comment by beckermr** [comment]
Do we? I built the vc feedstock on the current image.

**Comment by beckermr** [comment]
https://github.com/conda-forge/conda-smithy/blob/master/conda_smithy/templates/azure-pipelines-win.yml.tmpl#L8

What version do we need? We are at `vs2017-win2016`

**Comment by isuruf** [comment]
I meant appveyor. azure is fine.

**Comment by beckermr** [comment]
Available images are here:

https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops

looks like we could bump to windows-2019, but I don't understand why.

**Comment by beckermr** [comment]
appveyor is deprecated as of yesterday

**Comment by beckermr** [comment]
but I will make a PR

**Comment by isuruf** [comment]
> appveyor is deprecated as of yesterday

Isn't that for conda-forge feedstocks? I thought it wasn't deprecated for outside orgs.

**Comment by beckermr** [comment]
Not for outside orgs, but I didn't realize they used our pinnings. I will bump it.

**Comment by beckermr** [comment]
https://github.com/conda-forge/conda-smithy/pull/1278

**Comment by beckermr** [comment]
Alright @isuruf, let's merge?

**Pull request merged**